packages/server-core/src/projects/project/project-helper.ts

```diff
@@ -1751,7 +1751,7 @@ export const uploadLocalProjectToProvider = async (
   const resourceDBPath = path.join(projectRootPath, 'resources.json')
   const hasResourceDB = fs.existsSync(resourceDBPath)
 
-  const files = getFilesRecursive(projectRootPath)
+  const files = getFilesRecursive(projectRootPath, false, [path.join(projectRootPath, 'thumbnails')])
```

Comment:
rather than overloading getFilesRecursive with an exclusion, we should just filter the results once we have them. the overhead is negligible